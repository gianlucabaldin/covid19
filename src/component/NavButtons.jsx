/* eslint-disable import/no-cycle */
import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import { Box } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import { SECTIONS } from '../utils/consts';

const NavButtons = ({ onClick }) => {
  const { t } = useTranslation();
  const [active, setActive] = useState(SECTIONS.ITALY);

  const handleClick = (location) => {
    setActive(location);
    onClick(location);
  };

  return (
    <Box
      display="flex"
      justifyContent="center"
      m={1}
      p={1}
      data-test-id="nav-buttons"
    >
      <Box m={1}>
        <Button
          variant={active === SECTIONS.ITALY ? 'contained' : 'outlined'}
          color="primary"
          data-test-id="nav-button-italy"
          onClick={() => handleClick(SECTIONS.ITALY)}
        >
          {t('navbuttons.italy')}
        </Button>
      </Box>
      <Box m={1}>
        <Button
          variant={active === SECTIONS.EUROPE ? 'contained' : 'outlined'}
          color="primary"
          data-test-id="nav-button-europe"
          onClick={() => handleClick(SECTIONS.EUROPE)}
        >
          {t('navbuttons.europe')}
        </Button>
      </Box>

      <Box m={1}>
        <Button
          variant={active === SECTIONS.WORLDWIDE ? 'contained' : 'outlined'}
          color="primary"
          data-test-id="nav-button-worldwide"
          onClick={() => handleClick(SECTIONS.WORLDWIDE)}
        >
          {t('navbuttons.worldwide')}
        </Button>
      </Box>
    </Box>
  );
};

export default NavButtons;
