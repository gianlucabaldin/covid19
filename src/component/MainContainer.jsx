/* eslint-disable react/jsx-props-no-spreading */
import React, { useState, useEffect } from 'react';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import NavButtons from './NavButtons';
import Summary from './Summary';
import ChartItaly from './ChartItaly';
import { mockResponseJson } from './mockResponse';

export const SECTIONS = {
  WORLDWIDE: 0,
  ITALY: 1,
  COUNTRY_LIST: 2,
};

const initialStatus = {
  confirmed: 0,
  recovered: 0,
  deaths: 0,
  error: false,
};

const initialChartStatus = {
  data: {},
  error: false,
};

const MainContainer = () => {
  const [summaryData, setSummaryData] = useState({ ...initialStatus });
  const [chartData, setChartData] = useState({ ...initialChartStatus });
  // const [activeSection, setActiveSection] = useState(SECTIONS.ITALY);

  // WORKING -- UNCOMMENT WHEN FINISHED
  /*
  const fetchData = async () => {
    const response = await fetch(
      'https://api.covid19api.com/total/dayone/country/italy',
    );
    response
      .json()
      .then((res) => {
        setChartData({ data: res });
        const { Confirmed, Recovered, Deaths } = res[res.length - 1];
        setSummaryData({
          confirmed: Confirmed,
          recovered: Recovered,
          deaths: Deaths,
          error: false,
        });
      })
      .catch((err) => {
        console.log(err);

        setSummaryData({ error: true });
        setChartData({ error: true });
      });
  };
  useEffect(() => {
    fetchData();
    // }, [activeSection]);
  }, []);
*/

  const mock = mockResponseJson;

  return (
    <Container fixed>
      <NavButtons />
      {/* <Typography
        component="div"
        style={{ backgroundColor: '#CFE8FC', height: '100vh' }}
      > */}
      {/* <Summary {...summaryData} /> */}
      {/* <ChartItaly {...chartData} /> */}
      <ChartItaly data={mock} />
      {/* </Typography> */}
    </Container>
  );
};

export default MainContainer;
