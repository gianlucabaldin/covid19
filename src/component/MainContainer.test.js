import React from 'react';
import { shallow } from 'enzyme';
import MainContainer from './MainContainer';
import NavButtons from './NavButtons';

describe('MainContainer', () => {
  const wrapper = shallow(<MainContainer />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('contains NavButtons component', () => {
    expect(wrapper).toContainReact(<NavButtons />);
  });
  /*
  it('has a left-aligned `Covid 19` text', () => {
    expect(wrapper.find('.MuiTypography-h6').exists()).toEqual(true);
    expect(wrapper).toIncludeText('Covid 19');
  });

  it('has a center-aligned project description - with Enzyme', () => {
    const { getByText } = render(<Header />);
    expect(getByText('Latest updates & historical data')).toBeInTheDocument();
  });

  // https://github.com/FormidableLabs/enzyme-matchers/blob/master/packages/jest-enzyme/README.md
  it('has a center-aligned project description - with jest-enzyme', () => {
    expect(wrapper).toIncludeText('Latest updates & historical data');
  }); */
});
