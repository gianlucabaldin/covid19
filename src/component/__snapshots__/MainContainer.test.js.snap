// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer renders properly 1`] = `
<WithStyles(ForwardRef(Container))
  fixed={true}
>
  <NavButtons />
  <WithStyles(ForwardRef(Typography))
    component="div"
    style={
      Object {
        "backgroundColor": "#CFE8FC",
        "height": "100vh",
      }
    }
  >
    <Summary
      confirmed={0}
      data={
        Object {
          "confirmed": 0,
          "deaths": 0,
          "error": false,
          "recovered": 0,
        }
      }
      deaths={0}
      error={false}
      recovered={0}
    />
    <ChartItaly
      data={Object {}}
      error={false}
    />
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(Container))>
`;
